lib/text_diff.ml

```diff
@@ -119,24 +126,229 @@ let process_atomic_diff (diff : (_, _ Diff.atomic_modification) Diff.t) name
+                ];
+            };
+        ]
+
+and pair_of_privacy_diff privacy_diff =
+  match privacy_diff with
+  | Added_p -> (Asttypes.Public, Asttypes.Private)
+  | Removed_p -> (Asttypes.Private, Asttypes.Public)
+
+and pair_of_manifest_diff manifest_diff =
+  match manifest_diff with
+  | Added te -> (None, Some te)
+  | Removed te -> (Some te, None)
+  | Modified { reference; current } -> (Some reference, Some current)
+
+and type_header_to_line name private_flag type_expr_opt abstract =
+  let private_flag_str = string_of_private_flag private_flag in
+  let manifest_str = string_of_manifest type_expr_opt in
+  let equal_sign =
+    if abstract && private_flag_str = "" && manifest_str = "" then "" else " ="
+  in
+  Printf.sprintf "type %s%s%s%s" name equal_sign private_flag_str manifest_str
+
+and string_of_private_flag private_flag =
+  match private_flag with
+  | Asttypes.Public -> ""
+  | Asttypes.Private -> " private"
+
+and string_of_manifest manifest =
+  match manifest with None -> "" | Some te -> " " ^ typ_expr_to_line te
+
+and process_type_kind_diff type_kind =
+  match type_kind with
+  | Either.Left same_type_kind ->
+      `Same
+        [
+          Same
+            (String.concat "\n"
+               (Option.value (type_kind_to_lines same_type_kind) ~default:[]));
+        ]
+  | Either.Right (Record_tk changed_lst) ->
+      `Compound_change
+        (process_modified_record_type_diff ~indent_amount:1 changed_lst)
+  | Either.Right (Variant_tk changed_lst) ->
+      `Compound_change (process_modified_variant_type_diff changed_lst)
+  | Either.Right (Atomic_tk { reference; current }) ->
+      `Atomic_change
+        [
+          Change
+            {
+              orig = Option.value (type_kind_to_lines reference) ~default:[];
+              new_ = [];
+            };
+          Change
+            {
+              orig = [];
+              new_ = Option.value (type_kind_to_lines current) ~default:[];
+            };
+        ]
+
+and type_kind_to_lines type_kind : string list option =
+  match type_kind with
+  | Types.Type_record (lbl_lst, _) ->
+      Some
+        [
+          indent_s 1 "{ "
+          ^ (List.map lbl_to_line lbl_lst |> String.concat " ")
+          ^ " }";
+        ]
+  | Types.Type_variant (cstr_lst, _) ->
+      Some
+        (List.map
+           (fun s -> indent_s 1 s)
+           (List.concat_map (fun cstr -> cstr_to_lines cstr) cstr_lst))
+  | Types.Type_abstract _ -> None
+  | Types.Type_open -> Some [ indent_s 1 ".." ]
+
+and indent_s n s =
+  let indentation = String.init n (fun _ -> ' ') in
+  indentation ^ s
+
+and process_modified_record_type_diff ~indent_amount diff =
   let changes = process_modified_labels diff in
-  [ indent indent_n (Same "...") ]
-  @ List.map (fun c -> indent indent_n c) changes
+  [ indent indent_amount (Same "{") ]
+  @ [ indent (indent_amount + 2) (Same "...") ]
+  @ List.map (fun c -> indent (indent_amount + 2) c) changes
```

Comment:
You don't need to construct a singleton list and concatenate here, you can simply append:
```suggestion
  (indent indent_amount (Same "{"))
  :: (indent (indent_amount + 2) (Same "..."))
  :: (List.map (fun c -> indent (indent_amount + 2) c) changes)
```